sync-core/src/main/java/com/cloudant/sync/query/IndexCreator.java

```diff
@@ -158,23 +156,41 @@ public Boolean call() {
                     }
                 }
 
-                String statement = null;
-                try {
-                    List<String> columnList = new ArrayList<String>();
-                    for (String field: fieldNamesList) {
-                        columnList.add("\"" + field + "\"");
+                // Create SQLite data structures to support the index
+                // For JSON index type create a SQLite table and a SQLite index
+                // For TEXT index type create a SQLite virtual table
+                List<String> columnList = new ArrayList<String>();
+                for (String field: fieldNamesList) {
+                    columnList.add("\"" + field + "\"");
+                }
+
+                List<String> statements = new ArrayList<String>();
+                if (index.indexType.equalsIgnoreCase(Index.TEXT_TYPE)) {
+                    List<String> settingsList = new ArrayList<String>();
+                    // Ensure that the _id and _rev columns are not included
+                    // in the text index.
+                    settingsList.add("notindexed=\"_id\"");
+                    settingsList.add("notindexed=\"_rev\"");
```

Comment:
This means that when doing a full text search on a term, that term will not be applied to the `_id` or the `_rev` columns.